### PR TITLE
Metadata show partial + basic default mods rendering

### LINKED ISSUE #883: Add a button to display a modal that shows html mods (adding a route to support non-JS)
Design in #799 

mock

 - [x] Button on show page that opens up modal with mods xml display page
 - [x] Button is a link to mods xml display page with no js

See Blacklight `add_show_tools_partial`
https://github.com/projectblacklight/blacklight/wiki/Adding-new-document-actions

### PR DESCRIPTION
**Pull request by mejackreed**
Closes #883 
Also updates Blacklight to get a JS fix we needed.
Fixture.. anything :)

This only feature flags the button but not the route (I think thats ok :) )

mdbutton

## UPDATE!!

Added the basic metadata html

mdstuffs

Addresses #844.

I think additional changes / designs should be new issues.

### CODE REVIEW

File: app/views/catalog/_metadata_button_default.html.erb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<% if feature_flags.view_all_metadata? %>
```
Comment (cbeer) [inline app/views/catalog/_metadata_button_default.html.erb]: Now that we have the MODS display stuff working, do we need the feature flag still?

Comment (mejackreed) [reply app/views/catalog/_metadata_button_default.html.erb]: Great idea. I've pulled that out now.

File: config/routes.rb [hunk 2]
```diff
@@ -28,11 +28,20 @@
...
   mount Spotlight::Engine, at: '/'
+  concern :exportable, Blacklight::Routes::Exportable.new
 
   resources :exhibits, path: '/', only: [] do
     resource :dor_harvester, controller: :"dor_harvester", only: [:create, :update]
     resource :bibliography_resources, only: [:create, :update]
     resource :viewers, only: [:create, :edit, :update]
+
+    resources :solr_documents, except: [:index], path: '/catalog', controller: 'catalog' do
+      concerns :exportable
```
Comment (cbeer) [inline config/routes.rb]: I'm a little surprised we need to re-mix-in `exportable`. What do you see without it?

File: config/routes.rb [hunk 3]
```diff
@@ -28,11 +28,20 @@
...
 
   mount Spotlight::Engine, at: '/'
+  concern :exportable, Blacklight::Routes::Exportable.new
 
   resources :exhibits, path: '/', only: [] do
     resource :dor_harvester, controller: :"dor_harvester", only: [:create, :update]
     resource :bibliography_resources, only: [:create, :update]
     resource :viewers, only: [:create, :edit, :update]
+
+    resources :solr_documents, except: [:index], path: '/catalog', controller: 'catalog' do
```
Comment (cbeer) [inline config/routes.rb]: Instead of overriding the normal routes we're getting from spotlight, would `only: []` work (instead of `except: [:index]`)?

Comment (mejackreed) [reply config/routes.rb]: Yep! Added.. thanks

File: config/routes.rb (hunk 2 above)
Comment (mejackreed) [reply config/routes.rb]: `No route matches..` and `undefined method `exhibit_root_path' for #<Spotlight::CatalogController:0x007ff41d7be2b0> Did you mean? exhibit_viewers_path`

**Pull request merged**